src/utils/encoding.ts

```diff
@@ -158,19 +150,20 @@ export function parseWallet(wallet: VaultRegistryWallet, network: Network): Wall
     };
 }
 
-export function parseSystemVault(
+export async function parseSystemVault(
+    assetRegistryApi: AssetRegistryAPI,
     vault: VaultRegistrySystemVault,
     wrappedCurrency: WrappedCurrency,
-    collateralCurrency: CollateralCurrency
-): SystemVaultExt {
+    collateralCurrency: CollateralCurrencyExt
+): Promise<SystemVaultExt> {
     return {
         toBeIssuedTokens: newMonetaryAmount(vault.toBeIssuedTokens.toString(), wrappedCurrency),
         issuedTokens: newMonetaryAmount(vault.issuedTokens.toString(), wrappedCurrency),
         toBeRedeemedTokens: newMonetaryAmount(vault.toBeRedeemedTokens.toString(), wrappedCurrency),
         collateral: newMonetaryAmount(vault.collateral.toString(), collateralCurrency),
         currencyPair: {
-            collateralCurrency: currencyIdToMonetaryCurrency(vault.currencyPair.collateral),
-            wrappedCurrency: currencyIdToMonetaryCurrency(vault.currencyPair.wrapped),
+            collateralCurrency: await currencyIdToMonetaryCurrency(assetRegistryApi, vault.currencyPair.collateral),
+            wrappedCurrency: await currencyIdToMonetaryCurrency(assetRegistryApi, vault.currencyPair.wrapped),
```

Comment:
Minor nit: probably could await both promises concurrently

Reply:
> Minor nit: probably could await both promises concurrently

Good shout, will add to follow-up task. Thank you.